cmake/FindUnwind.cmake

```diff
@@ -0,0 +1,56 @@
+# FindUnwind.cmake - Find libunwind package.
+#
+# Copyright 2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+
+#-----------------------------------------------------------------------
+# Use pkg-config to search for the module.
+#-----------------------------------------------------------------------
+find_package(PkgConfig)
+
+pkg_check_modules(UNWIND QUIET libunwind)
+
+#-----------------------------------------------------------------------
+# Find libraries and include directories.
+#-----------------------------------------------------------------------
+find_path(UNWIND_INCLUDE_DIR
+    NAMES "unwind.h"
+    PATHS ${UNWIND_INCLUDE_DIRS}
+)
+
+find_library(UNWIND_LIBRARY
+    NAMES unwind
+    PATHS ${UNWIND_LIBRARY_DIRS}
+)
+
+#-----------------------------------------------------------------------
+# Get version number
+#-----------------------------------------------------------------------
+if(UNWIND_VERSION)
+    set(UNWIND_VERSION ${UNWIND_VERSION} CACHE STRING "libunwind version")
+endif()
+
+#-----------------------------------------------------------------------
+# Handle REQUIRED and QUIETLY arguments
+#-----------------------------------------------------------------------
+find_package_handle_standard_args(UNWIND
+    REQUIRED_VARS
+        UNWIND_LIBRARY
+        UNWIND_INCLUDE_DIR
+    VERSION_VAR
+        UNWIND_VERSION
+)
+
+mark_as_advanced(UNWIND_INCLUDE_DIR UNWIND_LIBRARY UNWIND_VERSION)
```

Comment:
My intention is not to _force_ the variable to be advanced; it is to mark it as advanced _by default_, allowing the author of the CMake script that issued the `find_package` command the final say ("if a variable already has an advanced/non-advanced state, it will not be changed"). I am doing exactly what the documentation says to do, for exactly the reason the documentation offers for doing it this way.

The three Find modules I'm submitting (https://github.com/ipdk-io/networking-recipe/pull/174, https://github.com/ipdk-io/networking-recipe/pull/189, and https://github.com/ipdk-io/networking-recipe/pull/190) are based on a combination of the CMake documentation, _Professional CMake_ by Craig Scott, and the source code for several existing Find modules.